packages/firestore/e2e/Aggregate/AggregateQuery.e2e.js

```diff
@@ -0,0 +1,272 @@
+      };
+
+      try {
+        getAggregateFromServer(colRef, aggregateSpec2);
+        return Promise.reject(new Error('Did not throw an Error.'));
+      } catch (error) {
+        error.message.should.containEql("'AggregateField' has an an unknown 'AggregateType'");
+      }
+      return Promise.resolve();
+    });
+  });
+
+  describe('count(), average() & sum()', function () {
+    it('no existing collection responses for average(), sum() & count()', async function () {
+      const { getAggregateFromServer, collection, getFirestore, count, average, sum } =
+        firestoreModular;
+      const firestore = getFirestore();
+
+      const colRefNoDocs = collection(firestore, `${COLLECTION}/aggregate-count/no-docs`);
+
+      const aggregateSpecNoDocuments = {
+        countCollection: count(),
+        averageBar: average('bar'),
+        sumBaz: sum('baz'),
+      };
+
+      const resultNoDocs = await getAggregateFromServer(colRefNoDocs, aggregateSpecNoDocuments);
+
+      const dataNoDocs = resultNoDocs.data();
+
+      // average returns null, whilst sum and count return 0
+      dataNoDocs.countCollection.should.eql(0);
+      should(dataNoDocs.averageBar).be.null();
+      dataNoDocs.sumBaz.should.eql(0);
+    });
+
+    it('single path using `string`', async function () {
+      const { getAggregateFromServer, doc, setDoc, collection, getFirestore, count, average, sum } =
+        firestoreModular;
+      const firestore = getFirestore();
+
+      const colRef = collection(firestore, `${COLLECTION}/aggregate-count/collection`);
+
+      await Promise.all([
+        setDoc(doc(colRef, 'one'), { bar: 5, baz: 4 }),
+        setDoc(doc(colRef, 'two'), { bar: 5, baz: 4 }),
+        setDoc(doc(colRef, 'three'), { bar: 5, baz: 4 }),
+      ]);
+
+      const aggregateSpec = {
+        ignoreThisProperty: 'not aggregate field',
+        countCollection: count(),
+        averageBar: average('bar'),
+        sumBaz: sum('baz'),
+      };
+
+      const result = await getAggregateFromServer(colRef, aggregateSpec);
+
+      const data = result.data();
+
+      data.countCollection.should.eql(3);
+      data.averageBar.should.eql(5);
+      data.sumBaz.should.eql(12);
+      // should only return the aggregate field requests
+      data.should.not.have.property('ignoreThisProperty');
+    });
+
+    it('single path using `FieldPath`', async function () {
+      const {
+        getAggregateFromServer,
+        doc,
+        setDoc,
+        collection,
+        getFirestore,
+        count,
+        average,
+        sum,
+        FieldPath,
+      } = firestoreModular;
+      const firestore = getFirestore();
+
+      const colRef = collection(firestore, `${COLLECTION}/aggregate-count-field-path/collection`);
+
+      await Promise.all([
+        setDoc(doc(colRef, 'one'), { bar: 5, baz: 4 }),
+        setDoc(doc(colRef, 'two'), { bar: 5, baz: 4 }),
+        setDoc(doc(colRef, 'three'), { bar: 5, baz: 4 }),
+      ]);
```

Comment:
I saw this and it reminded me of this incredibly traumatic issue https://github.com/invertase/react-native-firebase/issues/3004 😆 

It makes me want to poke at things a bit more to make sure we handle things correctly as javascript number handling is really picky. A classic is that 0.1 is not represented well (because it's IEEE floating point internally, not an integer or decimal)

Suggest making the average actually have to work by using 4 5 and 6
Suggest attempting an sum of 0.1, 0.1 and 0.1 to see what happens (is it 0.3, or is it 0.30000000000004? This is a javascript classic)

There are also firestore limitations where values that not numbers are ignored during the aggregation but since that is all on the backend I don't think we need to probe those sorts of limitations - but verifying our number handling (is it a "number", or an "integer" or a "double" ?) may be insightful

Reply:
Ah ha!! Not sure exactly what to do about this but the PR comments from the one that resolved #3004 may be instructive? This is really subtle stuff. The hunch was right though - check the result on iOS:

>      AssertionError: expected 0.30000000000000004 to equal 0.3

Devious.

Reply:
Looks like all platforms reproduce the number-handling issue now, so it is consistent at least
Sorry my android push was poorly tested and had an immediate problem - had an issue running e2e android locally so just pushed without testing, and of course that's how an untested push goes...